pattlog.go

```diff
@@ -41,22 +53,26 @@ func FormatLogRecord(format string, rec *LogRecord) string {
 	}
 
 	out := bytes.NewBuffer(make([]byte, 0, 64))
-	secs := rec.Created.UnixNano() / 1e9
+	millis := rec.Created.UnixNano() / 1e6
+	seconds := millis / 1000
 
-	cache := *formatCache
-	if cache.LastUpdateSeconds != secs {
-		month, day, year := rec.Created.Month(), rec.Created.Day(), rec.Created.Year()
-		hour, minute, second := rec.Created.Hour(), rec.Created.Minute(), rec.Created.Second()
+	// Check if we need to recompute any caches (has it been 1 second since we last logged?)
+	hour, minute, second := rec.Created.Hour(), rec.Created.Minute(), rec.Created.Second()
+	if timeFormatCache.lastSecond != seconds {
 		zone, _ := rec.Created.Zone()
-		updated := &formatCacheType{
-			LastUpdateSeconds: secs,
-			shortTime:         fmt.Sprintf("%02d:%02d", hour, minute),
-			shortDate:         fmt.Sprintf("%02d/%02d/%02d", month, day, year%100),
-			longTime:          fmt.Sprintf("%02d:%02d:%02d %s", hour, minute, second, zone),
-			longDate:          fmt.Sprintf("%04d/%02d/%02d", year, month, day),
+		timeFormatCache.lastSecond = seconds
+		timeFormatCache.longTime = fmt.Sprintf("%02d:%02d:%02d %s", hour, minute, second, zone)
+		timeFormatCache.shortTime = fmt.Sprintf("%02d:%02d", hour, minute)
+
+		// Check if we need to recompute the date cache
+		month, day, year := rec.Created.Month(), rec.Created.Day(), rec.Created.Year()
+		if dateFormatCache.lastDay != day || dateFormatCache.lastMonth != int(month) || dateFormatCache.lastYear != year {
```

Comment:
Is it always true that we would only ever be able to invalidate the date cache if the second cache was also invalidated?  Thinking things like leap seconds and the like.  It might be safer to pull the check out of the `if timeFormatCache.lastSecond != seconds {` conditional to be safe.  It doesn't *need* to be in there as far as I can tell.